### ISSUE TITLE
wasm_exec.js: out-of-sync with runtime.walltime changes

### ISSUE THREAD

**Issue by flyn-org** [golang/go#52900]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.18 linux/amd64
</pre>

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE=".../.cache/go-build"
GOENV=".../.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE=".../go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH=".../go"
GOPRIVATE=""
GOPROXY="direct"
GOROOT="/usr/lib/golang"
GOSUMDB="off"
GOTMPDIR=""
GOTOOLDIR="/usr/lib/golang/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.18"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=".../go.mod"
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build4127447452=/tmp/go-build -gno-record-gcc-switches"
</pre></details>

### What did you do?

I maintain a number of applications that use Go's WebAssembly support. A recent compiler update broke things. I am using the wasm_exec.js file that was shipped with my updated compiler, but my browser prints the following error to its console:
```
Uncaught (in promise) LinkError: import object field 'runtime.walltime' is not a Function 
```
Might this be related to https://groups.google.com/g/golang-checkins/c/Mhnkp2QrdBo? I did see `walltime` mentioned in `src/runtime/sys_wasm.s`.

**Comment by flyn-org** [comment]
Everything works when I revert to Go 1.16.8 and recompile my application.

**Comment by dmitshur** [comment]
To clarify, are you running into this when using a .wasm module built with Go 1.18 together with the wasm_exec.js file from Go 1.18? Are you able to reproduce it with a simple hello-world program, or does it not reproduce on such a small program?

**Comment by flyn-org** [comment]
Oddly, things fixed when I returned to 1.18 for further testing. I expect I overlooked something obvious, but I had tested Go 1.18 with the Go 1.18 wasm_exec.js file. What was broken now works again.

I will close this for now. I will be updating some of my other applications in the next few days, and I will reopen this if I find it necessary. For now, it looks like I made a mistake somewhere.
